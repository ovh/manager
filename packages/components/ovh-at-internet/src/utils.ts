export function debug(...args: any[]) {
  if (!window.test && //temporarily  fix jest bug
    window.localStorage?.getItem('MANAGER_TRACKING_DEBUG')) {
    console.debug(...args);
  }
}
