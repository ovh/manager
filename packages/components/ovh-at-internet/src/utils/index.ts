export const getRandomString = (len: number): string => {
  const alphabet =
    'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789';
  return new Array(len)
    .map(() => alphabet.charAt(Math.floor(Math.random() * alphabet.length)))
    .join('');
};

/**
 * Returns an unique ID (used by orders and carts)
 * To ensure that the ID is unique, we generate it from current milliseconds
 * and concat a random string to avoid any collisions.
 */
export const getUniqueId = (): string => {
  // unique key : current milliseconds in base 36 concatenated with 8 random chars
  return new Date().valueOf().toString(36) + getRandomString(8);
};
