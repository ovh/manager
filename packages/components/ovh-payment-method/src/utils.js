/**
 * Checks if a value is strictly null.
 * @param value - The value to check.
 * @returns Whether the value is null.
 */
export const isNull = (value) => value === null;
