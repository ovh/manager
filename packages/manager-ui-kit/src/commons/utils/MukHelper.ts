export function hashCode(value: unknown): number {
  const str = (() => {
    switch (typeof value) {
      case 'number':
      case 'bigint':
      case 'boolean':
      case 'string':
        return String(value);
      case 'object':
        return value === null ? 'null' : JSON.stringify(value);
      case 'function':
        return 'function';
      case 'undefined':
        return 'undefined';
      default:
        return 'symbol';
    }
  })();

  let hash = 0;
  for (let i = 0; i < str.length; i++) {
    hash = (hash * 32 - hash + str.charCodeAt(i)) % 2147483647;
  }

  return hash;
}
