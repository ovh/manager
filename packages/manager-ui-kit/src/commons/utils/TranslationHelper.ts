import i18next from 'i18next';
import type { Resource, ResourceLanguage } from 'i18next';

import { locales } from '@/commons/settings/Locales.constants';

/**
 * Manager Fallback Language
 */
const FALLBACK_LANGUAGE = 'fr_FR';

/**
 * Normalizes a language code to a format compatible with i18next loaders.
 * Converts hyphenated codes to underscore format and ensures region codes exist.
 *
 * Examples:
 * - 'fr-FR' → 'fr_FR'
 * - 'en' → 'en_GB'
 * - 'de' → 'de_DE'
 *
 * @param language - The language code from browser or user input.
 * @returns Normalized language code in the format `ll_RR` (e.g., 'en_GB').
 */
export const normalizeLanguageCode = (language: string): string => {
  let normalizedLang = language.replace('-', '_');

  if (!normalizedLang.includes('_')) {
    normalizedLang =
      normalizedLang === 'en' ? 'en_GB' : `${normalizedLang}_${normalizedLang.toUpperCase()}`;
  }

  return normalizedLang;
};

/**
 * Builds a resource loader function for dynamic translation loading.
 *
 * @param translationLoaders - A map of language codes to dynamic import functions for translations.
 * @param namespace - The namespace under which translations will be added.
 */
export const buildTranslationResources =
  (
    translationLoaders: Record<
      string,
      () => Promise<{ default?: Resource; [key: string]: unknown }>
    >,
    namespace: string,
  ) =>
  async (language: string): Promise<boolean> => {
    const normalizedLang = normalizeLanguageCode(language);

    // Always load fallbackLang once, before the user language
    if (!i18next.hasResourceBundle(FALLBACK_LANGUAGE, namespace)) {
      const fallbackLoader = translationLoaders[FALLBACK_LANGUAGE];
      if (fallbackLoader) {
        try {
          console.info(
            `Loading fallback language: ${FALLBACK_LANGUAGE} for namespace: ${namespace}`,
          );
          const fallbackModule = await fallbackLoader();
          const resources = (fallbackModule.default ?? fallbackModule) as Resource;
          i18next.addResources(FALLBACK_LANGUAGE, namespace, resources);
        } catch (error) {
          console.error(`Failed to load fallback translations (${FALLBACK_LANGUAGE}):`, error);
        }
      }
    }

    // Then load the requested language
    if (
      normalizedLang !== FALLBACK_LANGUAGE &&
      !i18next.hasResourceBundle(normalizedLang, namespace)
    ) {
      const loader = translationLoaders[normalizedLang];
      if (loader) {
        try {
          const module = await loader();
          const resources = (module.default ?? module) as Resource;
          i18next.addResources(normalizedLang, namespace, resources);
        } catch (error) {
          console.warn(
            `Could not load ${normalizedLang}. Falling back to ${FALLBACK_LANGUAGE}.`,
            error,
          );
        }
      }
    }

    return true;
  };

/**
 * Initializes and manages i18next language updates and resource loading.
 * Listens to i18next events and browser language changes to update translations dynamically.
 *
 * @param translationLoaders - A map of language codes to dynamic import functions.
 * @param namespace - The namespace to be used for loading translations.
 */
export const buildTranslationManager = (
  translationLoaders: Record<string, () => Promise<{ default?: Resource; [key: string]: unknown }>>,
  namespace: string,
): void => {
  const loadTranslations = buildTranslationResources(translationLoaders, namespace);

  const handleLanguageChange = async (lang: string): Promise<void> => {
    const normalizedLang = normalizeLanguageCode(lang);
    await loadTranslations(normalizedLang);
    console.log('Language changed to:', normalizedLang);
  };

  // Ensure promises are properly handled for event-based hooks
  if (i18next.isInitialized) {
    void handleLanguageChange(i18next.language);
  } else {
    i18next.on('initialized', () => {
      void handleLanguageChange(i18next.language);
    });
  }

  i18next.on('languageChanged', (lang: string) => {
    void handleLanguageChange(lang);
  });
};

/**
 * Union type of supported locale identifiers.
 */
export type LocaleType = (typeof locales)[number];

/**
 * Asynchronous loader that returns an i18next-compatible language resource.
 */
export type TranslationLoaderType = () => Promise<ResourceLanguage>;

/**
 * Creates a resolver function that unwraps a dynamically imported module’s default export.
 *
 * @template T - Type of the resolved module content
 * @returns A function that converts `import('./file.json')` to a typed Promise<T>.
 *
 * @example
 * const resolveTranslationModule = createImportResolver<ResourceLanguage>();
 * const resource = await resolveTranslationModule(import('./Messages_fr_FR.json'));
 */
const createImportResolver =
  <T>() =>
  (p: Promise<{ default: unknown }>) =>
    p.then((m) => m.default as T);

/**
 * A specialized resolver for translation JSON modules,
 * producing i18next {@link ResourceLanguage} objects.
 */
export const resolveTranslationModule = createImportResolver<ResourceLanguage>();
