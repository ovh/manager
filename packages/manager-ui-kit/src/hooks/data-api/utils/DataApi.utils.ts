import type { ColumnSort } from '@tanstack/react-table';

import type { Filter } from '@ovh-ux/manager-core-api';

export const filterEquals = (a: Filter, b: Filter) =>
  a.key === b.key && a.value === b.value && a.comparator === b.comparator;

export const getSearchParamsObject = (search: URLSearchParams) =>
  Object.fromEntries([...search.entries()]);

export const parseSorting = (params: URLSearchParams, defaultSorting?: ColumnSort): ColumnSort => {
  const sorting: ColumnSort = {
    id: '',
    desc: false,
  };
  if (params.has('sort')) {
    const sortParam = params.get('sort');
    if (sortParam) {
      sorting.id = sortParam;
      if (params.has('sortOrder')) {
        sorting.desc = params.get('sortOrder') === 'desc';
      }
    }
  } else if (defaultSorting) {
    return defaultSorting;
  }
  return sorting;
};
