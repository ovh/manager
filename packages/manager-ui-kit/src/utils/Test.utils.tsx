import { ReactNode } from 'react';

import { ShellContext, ShellContextType } from '@ovh-ux/manager-react-shell-client';

import { render } from '../../setupTest';

export const mockTrackPage = vitest.fn();
export const mockGetEnvironment = vitest.fn();

export const shellContext = {
  shell: {
    environment: {
      getEnvironment: mockGetEnvironment,
    },
    tracking: {
      trackPage: mockTrackPage,
    },
    navigation: {
      navigateTo: vitest.fn(),
      reload: vitest.fn(),
    },
    ux: {
      hidePreloader: vitest.fn(),
    },
  },
};

export const renderWithContext = ({ children }: { children: ReactNode }) =>
  render(
    <ShellContext.Provider value={shellContext as unknown as ShellContextType}>
      {children}
    </ShellContext.Provider>,
  );
