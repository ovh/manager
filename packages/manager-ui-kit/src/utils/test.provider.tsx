import React, { ComponentType } from 'react';
import { I18nextProvider, initReactI18next } from 'react-i18next';
import i18n from 'i18next';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import '@testing-library/jest-dom';
import 'element-internals-polyfill';

i18n.use(initReactI18next).init({
  fallbackLng: 'fr_FR',
  interpolation: {
    escapeValue: false,
  },
  resources: {
    en: {
      'action-menu': {
        common_actions: 'Actions',
      },
    },
  },
});

export const queryClient = new QueryClient({
  defaultOptions: {
    mutations: {
      retry: false,
    },
    queries: {
      retry: false,
    },
  },
});

const Wrappers = ({ children }: { children: React.ReactElement }) => {
  return (
    <QueryClientProvider client={queryClient}>
      <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
    </QueryClientProvider>
  );
};

const customRender = (
  ui: React.JSX.Element,
  options?: Omit<RenderOptions, 'queries'>,
): RenderResult =>
  render(ui, { wrapper: Wrappers as ComponentType, ...options });

export { customRender as render };
