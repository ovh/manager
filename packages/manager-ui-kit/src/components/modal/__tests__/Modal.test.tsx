import { describe, it } from 'vitest';
import { MODAL_COLOR } from '@ovhcloud/ods-react';
import { renderModal, heading, actions, ModalContent } from './ModalTest.utils';

describe('Modal Snapshot Tests', () => {
  it('displays the basic modal', () => {
    const { baseElement } = renderModal({
      heading,
      children: <ModalContent />,
    });
    expect(baseElement).toMatchSnapshot();
  });

  it('displays the modal with actions', () => {
    const { baseElement } = renderModal({
      heading,
      children: <ModalContent />,
      ...actions,
    });
    expect(baseElement).toMatchSnapshot();
  });

  it('displays loading Modal', () => {
    const { baseElement } = renderModal({
      heading,
      children: <ModalContent />,
      ...actions,
      loading: true,
    });
    expect(baseElement).toMatchSnapshot();
  });

  it('displays disabled primary and secondary buttons', () => {
    const { baseElement } = renderModal({
      heading,
      children: <ModalContent />,
      ...{
        primaryButton: {
          ...actions.primaryButton,
          disabled: true,
        },
        secondaryButton: {
          ...actions.secondaryButton,
          disabled: true,
        },
      },
    });
    expect(baseElement).toMatchSnapshot();
  });

  it('displays loading primary and secondary buttons', () => {
    const { baseElement } = renderModal({
      heading,
      children: <ModalContent />,
      ...{
        primaryButton: {
          ...actions.primaryButton,
          loading: true,
        },
        secondaryButton: {
          ...actions.secondaryButton,
          loading: true,
        },
      },
    });
    expect(baseElement).toMatchSnapshot();
  });

  it('displays the modal with critical type', () => {
    const { baseElement } = renderModal({
      heading,
      children: <ModalContent />,
      ...actions,
      type: MODAL_COLOR.critical,
    });
    expect(baseElement).toMatchSnapshot();
  });
});
