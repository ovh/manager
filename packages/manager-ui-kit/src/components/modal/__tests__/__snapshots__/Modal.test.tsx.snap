// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Modal Snapshot Tests > displays disabled primary and secondary buttons 1`] = `
<body>
  <div />
  <div
    class="_modal-backdrop_inusz_22"
    data-part="backdrop"
    data-scope="dialog"
    data-state="open"
    dir="ltr"
    id="dialog::r3::backdrop"
  />
  <div
    class="_modal-positioner_inusz_29"
    data-part="positioner"
    data-scope="dialog"
    dir="ltr"
    id="dialog::r3::positioner"
  >
    <div
      aria-describedby=""
      aria-labelledby=""
      aria-modal="true"
      class="_modal-content_inusz_35"
      data-ods="modal-content"
      data-part="content"
      data-scope="dialog"
      data-state="open"
      dir="ltr"
      id="dialog::r3::content"
      role="dialog"
      tabindex="-1"
    >
      <div
        class="_modal-header_1qhu0_2 _modal-header--information_1qhu0_17"
      >
        <button
          aria-label="Close"
          class="_button_1koyw_2 _button--neutral_1koyw_83 _button--xs_1koyw_25 _button--ghost_1koyw_151 _modal-header__close_1qhu0_26"
          data-ods="button"
          data-part="close-trigger"
          data-scope="dialog"
          dir="ltr"
          id="dialog::r3::close"
          type="button"
        >
          <span
            class="_icon_10c23_2 _icon--xmark_10c23_562"
            data-ods="icon"
            role="presentation"
          />
        </button>
      </div>
      <div
        class="_modal-body_3om6u_2 text-left"
        data-ods="modal-body"
      >
        <div
          class="flex items-center mb-4"
        >
          <h4
            class="_text--heading-4_1aw48_31 block mr-3 flex-1"
            data-ods="text"
          >
            Example Heading
          </h4>
        </div>
        <div
          class="flex flex-col text-left"
        >
          <div>
            <p>
              Lorem ipsum dolor sit amet consectetur adipiscing elit. Quisque faucibus ex sapien vitae pellentesque sem placerat. In id cursus mi pretium tellus duis convallis. Tempus leo eu aenean sed diam urna tempor. Pulvinar vivamus fringilla lacus nec metus bibendum egestas. Iaculis massa nisl malesuada lacinia integer nunc posuere. Ut hendrerit semper vel class aptent taciti sociosqu. Ad litora torquent per conubia nostra inceptos himenaeos.
            </p>
            <label
              for="test-input"
            >
              Test Input
            </label>
            <input
              data-testid="test-input"
              id="test-input"
              type="text"
            />
          </div>
        </div>
        <div
          class="flex justify-end flex-wrap gap-4"
        >
          <button
            class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151 mt-4"
            data-ods="button"
            data-testid="secondary-button"
            disabled=""
            type="button"
          >
            Cancel
          </button>
          <button
            class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--default_1koyw_49 mt-4"
            data-ods="button"
            data-testid="primary-button"
            disabled=""
            type="button"
          >
            Confirm
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Modal Snapshot Tests > displays loading Modal 1`] = `
<body>
  <div />
  <div
    class="_modal-backdrop_inusz_22"
    data-part="backdrop"
    data-scope="dialog"
    data-state="open"
    dir="ltr"
    id="dialog::r2::backdrop"
  />
  <div
    class="_modal-positioner_inusz_29"
    data-part="positioner"
    data-scope="dialog"
    dir="ltr"
    id="dialog::r2::positioner"
  >
    <div
      aria-describedby=""
      aria-labelledby=""
      aria-modal="true"
      class="_modal-content_inusz_35"
      data-ods="modal-content"
      data-part="content"
      data-scope="dialog"
      data-state="open"
      dir="ltr"
      id="dialog::r2::content"
      role="dialog"
      tabindex="-1"
    >
      <div
        class="_modal-header_1qhu0_2 _modal-header--information_1qhu0_17"
      >
        <button
          aria-label="Close"
          class="_button_1koyw_2 _button--neutral_1koyw_83 _button--xs_1koyw_25 _button--ghost_1koyw_151 _modal-header__close_1qhu0_26"
          data-ods="button"
          data-part="close-trigger"
          data-scope="dialog"
          dir="ltr"
          id="dialog::r2::close"
          type="button"
        >
          <span
            class="_icon_10c23_2 _icon--xmark_10c23_562"
            data-ods="icon"
            role="presentation"
          />
        </button>
      </div>
      <div
        class="_modal-body_3om6u_2 text-left"
        data-ods="modal-body"
      >
        <div
          class="flex items-center mb-4"
        >
          <h4
            class="_text--heading-4_1aw48_31 block mr-3 flex-1"
            data-ods="text"
          >
            Example Heading
          </h4>
        </div>
        <div
          class="flex justify-center my-5"
          data-testid="spinner"
        >
          <span
            class="_spinner_o7c7y_10 _spinner--primary_o7c7y_29 _spinner--md_o7c7y_20"
            data-ods="spinner"
            role="progressbar"
          >
            <svg
              aria-hidden="true"
              class="_spinner__svg_o7c7y_32"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="24"
                width="24"
              />
              <g
                transform="translate(-237.694 -93.932)"
              >
                <path
                  d="M-5236.305-4108.069h0v-2a9.011,9.011,0,0,0,9-9,9.01,9.01,0,0,0-9-9,9.01,9.01,0,0,0-9,9h-2a10.928,10.928,0,0,1,3.222-7.779,10.928,10.928,0,0,1,7.778-3.221,10.929,10.929,0,0,1,7.779,3.221,10.932,10.932,0,0,1,3.221,7.777,10.928,10.928,0,0,1-3.221,7.778A10.931,10.931,0,0,1-5236.305-4108.069Z"
                  transform="translate(5486 4225)"
                />
                <circle
                  cx="1"
                  cy="1"
                  r="1"
                  transform="translate(248.694 114.932)"
                />
                <circle
                  cx="1"
                  cy="1"
                  r="1"
                  transform="translate(238.694 104.932)"
                />
              </g>
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Modal Snapshot Tests > displays loading primary and secondary buttons 1`] = `
<body>
  <div />
  <div
    class="_modal-backdrop_inusz_22"
    data-part="backdrop"
    data-scope="dialog"
    data-state="open"
    dir="ltr"
    id="dialog::r4::backdrop"
  />
  <div
    class="_modal-positioner_inusz_29"
    data-part="positioner"
    data-scope="dialog"
    dir="ltr"
    id="dialog::r4::positioner"
  >
    <div
      aria-describedby=""
      aria-labelledby=""
      aria-modal="true"
      class="_modal-content_inusz_35"
      data-ods="modal-content"
      data-part="content"
      data-scope="dialog"
      data-state="open"
      dir="ltr"
      id="dialog::r4::content"
      role="dialog"
      tabindex="-1"
    >
      <div
        class="_modal-header_1qhu0_2 _modal-header--information_1qhu0_17"
      >
        <button
          aria-label="Close"
          class="_button_1koyw_2 _button--neutral_1koyw_83 _button--xs_1koyw_25 _button--ghost_1koyw_151 _modal-header__close_1qhu0_26"
          data-ods="button"
          data-part="close-trigger"
          data-scope="dialog"
          dir="ltr"
          id="dialog::r4::close"
          type="button"
        >
          <span
            class="_icon_10c23_2 _icon--xmark_10c23_562"
            data-ods="icon"
            role="presentation"
          />
        </button>
      </div>
      <div
        class="_modal-body_3om6u_2 text-left"
        data-ods="modal-body"
      >
        <div
          class="flex items-center mb-4"
        >
          <h4
            class="_text--heading-4_1aw48_31 block mr-3 flex-1"
            data-ods="text"
          >
            Example Heading
          </h4>
        </div>
        <div
          class="flex flex-col text-left"
        >
          <div>
            <p>
              Lorem ipsum dolor sit amet consectetur adipiscing elit. Quisque faucibus ex sapien vitae pellentesque sem placerat. In id cursus mi pretium tellus duis convallis. Tempus leo eu aenean sed diam urna tempor. Pulvinar vivamus fringilla lacus nec metus bibendum egestas. Iaculis massa nisl malesuada lacinia integer nunc posuere. Ut hendrerit semper vel class aptent taciti sociosqu. Ad litora torquent per conubia nostra inceptos himenaeos.
            </p>
            <label
              for="test-input"
            >
              Test Input
            </label>
            <input
              data-testid="test-input"
              id="test-input"
              type="text"
            />
          </div>
        </div>
        <div
          class="flex justify-end flex-wrap gap-4"
        >
          <button
            class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151 mt-4"
            data-ods="button"
            data-testid="secondary-button"
            type="button"
          >
            <span
              class="_button__spinner_1koyw_373"
            >
              <span
                class="_spinner_o7c7y_10 _spinner--neutral_o7c7y_26 _spinner--sm_o7c7y_17"
                data-ods="spinner"
                role="none"
              >
                <svg
                  aria-hidden="true"
                  class="_spinner__svg_o7c7y_32"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="none"
                    height="24"
                    width="24"
                  />
                  <g
                    transform="translate(-237.694 -93.932)"
                  >
                    <path
                      d="M-5236.305-4108.069h0v-2a9.011,9.011,0,0,0,9-9,9.01,9.01,0,0,0-9-9,9.01,9.01,0,0,0-9,9h-2a10.928,10.928,0,0,1,3.222-7.779,10.928,10.928,0,0,1,7.778-3.221,10.929,10.929,0,0,1,7.779,3.221,10.932,10.932,0,0,1,3.221,7.777,10.928,10.928,0,0,1-3.221,7.778A10.931,10.931,0,0,1-5236.305-4108.069Z"
                      transform="translate(5486 4225)"
                    />
                    <circle
                      cx="1"
                      cy="1"
                      r="1"
                      transform="translate(248.694 114.932)"
                    />
                    <circle
                      cx="1"
                      cy="1"
                      r="1"
                      transform="translate(238.694 104.932)"
                    />
                  </g>
                </svg>
              </span>
            </span>
            Cancel
          </button>
          <button
            class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--default_1koyw_49 mt-4"
            data-ods="button"
            data-testid="primary-button"
            type="button"
          >
            <span
              class="_button__spinner_1koyw_373"
            >
              <span
                class="_spinner_o7c7y_10 _spinner--neutral_o7c7y_26 _spinner--sm_o7c7y_17"
                data-ods="spinner"
                role="none"
              >
                <svg
                  aria-hidden="true"
                  class="_spinner__svg_o7c7y_32"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="none"
                    height="24"
                    width="24"
                  />
                  <g
                    transform="translate(-237.694 -93.932)"
                  >
                    <path
                      d="M-5236.305-4108.069h0v-2a9.011,9.011,0,0,0,9-9,9.01,9.01,0,0,0-9-9,9.01,9.01,0,0,0-9,9h-2a10.928,10.928,0,0,1,3.222-7.779,10.928,10.928,0,0,1,7.778-3.221,10.929,10.929,0,0,1,7.779,3.221,10.932,10.932,0,0,1,3.221,7.777,10.928,10.928,0,0,1-3.221,7.778A10.931,10.931,0,0,1-5236.305-4108.069Z"
                      transform="translate(5486 4225)"
                    />
                    <circle
                      cx="1"
                      cy="1"
                      r="1"
                      transform="translate(248.694 114.932)"
                    />
                    <circle
                      cx="1"
                      cy="1"
                      r="1"
                      transform="translate(238.694 104.932)"
                    />
                  </g>
                </svg>
              </span>
            </span>
            Confirm
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Modal Snapshot Tests > displays the basic modal 1`] = `
<body>
  <div />
  <div
    class="_modal-backdrop_inusz_22"
    data-part="backdrop"
    data-scope="dialog"
    data-state="open"
    dir="ltr"
    id="dialog::r0::backdrop"
  />
  <div
    class="_modal-positioner_inusz_29"
    data-part="positioner"
    data-scope="dialog"
    dir="ltr"
    id="dialog::r0::positioner"
  >
    <div
      aria-describedby=""
      aria-labelledby=""
      aria-modal="true"
      class="_modal-content_inusz_35"
      data-ods="modal-content"
      data-part="content"
      data-scope="dialog"
      data-state="open"
      dir="ltr"
      id="dialog::r0::content"
      role="dialog"
      tabindex="-1"
    >
      <div
        class="_modal-header_1qhu0_2 _modal-header--information_1qhu0_17"
      >
        <button
          aria-label="Close"
          class="_button_1koyw_2 _button--neutral_1koyw_83 _button--xs_1koyw_25 _button--ghost_1koyw_151 _modal-header__close_1qhu0_26"
          data-ods="button"
          data-part="close-trigger"
          data-scope="dialog"
          dir="ltr"
          id="dialog::r0::close"
          type="button"
        >
          <span
            class="_icon_10c23_2 _icon--xmark_10c23_562"
            data-ods="icon"
            role="presentation"
          />
        </button>
      </div>
      <div
        class="_modal-body_3om6u_2 text-left"
        data-ods="modal-body"
      >
        <div
          class="flex items-center mb-4"
        >
          <h4
            class="_text--heading-4_1aw48_31 block mr-3 flex-1"
            data-ods="text"
          >
            Example Heading
          </h4>
        </div>
        <div
          class="flex flex-col text-left"
        >
          <div>
            <p>
              Lorem ipsum dolor sit amet consectetur adipiscing elit. Quisque faucibus ex sapien vitae pellentesque sem placerat. In id cursus mi pretium tellus duis convallis. Tempus leo eu aenean sed diam urna tempor. Pulvinar vivamus fringilla lacus nec metus bibendum egestas. Iaculis massa nisl malesuada lacinia integer nunc posuere. Ut hendrerit semper vel class aptent taciti sociosqu. Ad litora torquent per conubia nostra inceptos himenaeos.
            </p>
            <label
              for="test-input"
            >
              Test Input
            </label>
            <input
              data-testid="test-input"
              id="test-input"
              type="text"
            />
          </div>
        </div>
        <div
          class="flex justify-end flex-wrap gap-4"
        />
      </div>
    </div>
  </div>
</body>
`;

exports[`Modal Snapshot Tests > displays the modal with actions 1`] = `
<body>
  <div />
  <div
    class="_modal-backdrop_inusz_22"
    data-part="backdrop"
    data-scope="dialog"
    data-state="open"
    dir="ltr"
    id="dialog::r1::backdrop"
  />
  <div
    class="_modal-positioner_inusz_29"
    data-part="positioner"
    data-scope="dialog"
    dir="ltr"
    id="dialog::r1::positioner"
  >
    <div
      aria-describedby=""
      aria-labelledby=""
      aria-modal="true"
      class="_modal-content_inusz_35"
      data-ods="modal-content"
      data-part="content"
      data-scope="dialog"
      data-state="open"
      dir="ltr"
      id="dialog::r1::content"
      role="dialog"
      tabindex="-1"
    >
      <div
        class="_modal-header_1qhu0_2 _modal-header--information_1qhu0_17"
      >
        <button
          aria-label="Close"
          class="_button_1koyw_2 _button--neutral_1koyw_83 _button--xs_1koyw_25 _button--ghost_1koyw_151 _modal-header__close_1qhu0_26"
          data-ods="button"
          data-part="close-trigger"
          data-scope="dialog"
          dir="ltr"
          id="dialog::r1::close"
          type="button"
        >
          <span
            class="_icon_10c23_2 _icon--xmark_10c23_562"
            data-ods="icon"
            role="presentation"
          />
        </button>
      </div>
      <div
        class="_modal-body_3om6u_2 text-left"
        data-ods="modal-body"
      >
        <div
          class="flex items-center mb-4"
        >
          <h4
            class="_text--heading-4_1aw48_31 block mr-3 flex-1"
            data-ods="text"
          >
            Example Heading
          </h4>
        </div>
        <div
          class="flex flex-col text-left"
        >
          <div>
            <p>
              Lorem ipsum dolor sit amet consectetur adipiscing elit. Quisque faucibus ex sapien vitae pellentesque sem placerat. In id cursus mi pretium tellus duis convallis. Tempus leo eu aenean sed diam urna tempor. Pulvinar vivamus fringilla lacus nec metus bibendum egestas. Iaculis massa nisl malesuada lacinia integer nunc posuere. Ut hendrerit semper vel class aptent taciti sociosqu. Ad litora torquent per conubia nostra inceptos himenaeos.
            </p>
            <label
              for="test-input"
            >
              Test Input
            </label>
            <input
              data-testid="test-input"
              id="test-input"
              type="text"
            />
          </div>
        </div>
        <div
          class="flex justify-end flex-wrap gap-4"
        >
          <button
            class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151 mt-4"
            data-ods="button"
            data-testid="secondary-button"
            type="button"
          >
            Cancel
          </button>
          <button
            class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--default_1koyw_49 mt-4"
            data-ods="button"
            data-testid="primary-button"
            type="button"
          >
            Confirm
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Modal Snapshot Tests > displays the modal with critical type 1`] = `
<body>
  <div />
  <div
    class="_modal-backdrop_inusz_22"
    data-part="backdrop"
    data-scope="dialog"
    data-state="open"
    dir="ltr"
    id="dialog::r5::backdrop"
  />
  <div
    class="_modal-positioner_inusz_29"
    data-part="positioner"
    data-scope="dialog"
    dir="ltr"
    id="dialog::r5::positioner"
  >
    <div
      aria-describedby=""
      aria-labelledby=""
      aria-modal="true"
      class="_modal-content_inusz_35"
      data-ods="modal-content"
      data-part="content"
      data-scope="dialog"
      data-state="open"
      dir="ltr"
      id="dialog::r5::content"
      role="alertdialog"
      tabindex="-1"
    >
      <div
        class="_modal-header_1qhu0_2 _modal-header--critical_1qhu0_14"
      >
        <button
          aria-label="Close"
          class="_button_1koyw_2 _button--neutral_1koyw_83 _button--xs_1koyw_25 _button--ghost_1koyw_151 _modal-header__close_1qhu0_26"
          data-ods="button"
          data-part="close-trigger"
          data-scope="dialog"
          dir="ltr"
          id="dialog::r5::close"
          type="button"
        >
          <span
            class="_icon_10c23_2 _icon--xmark_10c23_562"
            data-ods="icon"
            role="presentation"
          />
        </button>
      </div>
      <div
        class="_modal-body_3om6u_2 text-left"
        data-ods="modal-body"
      >
        <div
          class="flex items-center mb-4"
        >
          <h4
            class="_text--heading-4_1aw48_31 block mr-3 flex-1"
            data-ods="text"
          >
            Example Heading
          </h4>
        </div>
        <div
          class="flex flex-col text-left"
        >
          <div>
            <p>
              Lorem ipsum dolor sit amet consectetur adipiscing elit. Quisque faucibus ex sapien vitae pellentesque sem placerat. In id cursus mi pretium tellus duis convallis. Tempus leo eu aenean sed diam urna tempor. Pulvinar vivamus fringilla lacus nec metus bibendum egestas. Iaculis massa nisl malesuada lacinia integer nunc posuere. Ut hendrerit semper vel class aptent taciti sociosqu. Ad litora torquent per conubia nostra inceptos himenaeos.
            </p>
            <label
              for="test-input"
            >
              Test Input
            </label>
            <input
              data-testid="test-input"
              id="test-input"
              type="text"
            />
          </div>
        </div>
        <div
          class="flex justify-end flex-wrap gap-4"
        >
          <button
            class="_button_1koyw_2 _button--critical_1koyw_49 _button--md_1koyw_42 _button--ghost_1koyw_151 mt-4"
            data-ods="button"
            data-testid="secondary-button"
            type="button"
          >
            Cancel
          </button>
          <button
            class="_button_1koyw_2 _button--critical_1koyw_49 _button--md_1koyw_42 _button--default_1koyw_49 mt-4"
            data-ods="button"
            data-testid="primary-button"
            type="button"
          >
            Confirm
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
